### PR TITLE
[No-8][Training]edit-and-delete-blog

### PR DESCRIPTION
**Pull request by nguyentienthinh**
Task name: Edit and delete blog
Document: https://docs.google.com/spreadsheets/d/1aWRDQ4TsW6BgTjm7iLPnI7R9VRPeZF0Fl6lEq-wfpa8/edit?folder=1KnNubuBhStnVllQ-62F3SW1mG3x9DT-G#gid=2142888040
Content need review: Review manager blog: Feature Edit and delete blog.
Checked with phpcs PSR-2, phpmd: success.

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -39,5 +39,8 @@
     Route::namespace('User')->group(function () {
         Route::get('/blog', 'BlogController@index')->name('blog.index');
         Route::post('/blog/stores', 'BlogController@store')->name('blog.store');
+        Route::get('/blog/delete/{id}', 'BlogController@destroy')->name('blog.destroy');
+        Route::get('/blog/edit/{id}', 'BlogController@edit')->name('blog.edit');
+        Route::post('/blog/update/{id}', 'BlogController@update')->name('blog.update');
```
Comment (loitq) [inline routes/web.php]: standard /blog/{param}/delete. 
https://laravel.com/docs/5.1/controllers#restful-resource-controllers

**Review by loitq — COMMENTED** [review]
i commented. please check

File: routes/web.php (hunk 1 above)
Comment (nguyentienthinh) [reply routes/web.php]: I fixed

**Review by loitq — APPROVED** [review]
LGTM.

File: app/Http/Controllers/User/BlogController.php [hunk 2]
```diff
@@ -23,7 +23,7 @@ public function defineUser() {
      */
     public function index()
     {
-        $blogs = Blog::all();
+        $blogs = Blog::where('user_id', '=', $this->defineUser()->id)->get();
```
Comment (ToaiLC) [inline app/Http/Controllers/User/BlogController.php]: $canDelete = $this->defineUser()->can_delete === Blog::IS_TRUE?
or format code

File: app/Http/Controllers/User/BlogController.php [hunk 3]
```diff
@@ -55,4 +55,53 @@ public function store(Request $request)
...
+
+    /**
+     * Delete one blog.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id) 
+    {
+        if ($this->defineUser()->can_delete === Blog::IS_TRUE) {
```
Comment (ToaiLC) [inline app/Http/Controllers/User/BlogController.php]: how do you know success case?

File: app/Http/Controllers/User/BlogController.php [hunk 4]
```diff
@@ -55,4 +55,53 @@ public function store(Request $request)
...
+            $blog = Blog::find($id);
```
Comment (ToaiLC) [inline app/Http/Controllers/User/BlogController.php]: what happen if blog empty?

File: app/Http/Controllers/User/BlogController.php [hunk 5]
```diff
@@ -55,4 +55,53 @@ public function store(Request $request)
...
+    {
+        $request->validate([
+            'title' => 'required',
+            'content' => 'required',
+        ]);
+        $blog = Blog::find($id);
+        $blog->title = $request->title;
+        $blog->content = $request->content;
+        $blog->save();
+        return redirect()->route('blog.index');
```
Comment (ToaiLC) [inline app/Http/Controllers/User/BlogController.php]: notify current state for user.

File: app/Http/Controllers/User/BlogController.php [hunk 6]
```diff
@@ -55,4 +55,53 @@ public function store(Request $request)
...
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request ,$id)
+    {
+        $request->validate([
+            'title' => 'required',
+            'content' => 'required',
+        ]);
```
Comment (ToaiLC) [inline app/Http/Controllers/User/BlogController.php:121]: add a empty line

File: app/Http/Controllers/User/BlogController.php (hunk 5 above)
Comment (ToaiLC) [inline app/Http/Controllers/User/BlogController.php]: add a empty line before return

File: resources/views/admin/layout/index.blade.php [hunk 7]
```diff
@@ -51,6 +51,9 @@
     <!-- SB Admin Scripts - Include with every page -->
     <script src="/asset/admin/js/sb-admin.js">
     </script>
+
+    <!-- ckeditor -->
+    <script src="{{ asset('ckeditor/ckeditor.js') }}"></script>
```
Comment (ToaiLC) [inline resources/views/admin/layout/index.blade.php]: format code
<script,,,>
</script>

File: app/Http/Controllers/User/BlogController.php (hunk 2 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: It can and i fixed

File: app/Http/Controllers/User/BlogController.php (hunk 3 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: i added notify

File: app/Http/Controllers/User/BlogController.php (hunk 4 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: I added error notify

File: app/Http/Controllers/User/BlogController.php (hunk 5 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: I added notify

Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: i fixed

File: resources/views/admin/layout/index.blade.php (hunk 7 above)
Comment (nguyentienthinh) [reply resources/views/admin/layout/index.blade.php]: i fixed

File: app/Http/Controllers/User/BlogController.php [hunk 8]
```diff
@@ -53,6 +49,65 @@ public function store(Request $request)
...
+        $blog = Blog::find($id);
+        if (isset($blog)) {
```
Comment (ToaiLC) [inline app/Http/Controllers/User/BlogController.php]: format code in block If

File: app/Http/Controllers/User/BlogController.php [hunk 9]
```diff
@@ -53,6 +49,65 @@ public function store(Request $request)
...
+        $blog->title = $request->title;
+        $blog->content = $request->content;
+        $blog->save();
+        } else {
```
Comment (ToaiLC) [inline app/Http/Controllers/User/BlogController.php]: Please apply code, for same block.
if (true) { blog-> save; return ....}
return failed;

File: resources/views/blog/index.blade.php [hunk 10]
```diff
@@ -1,69 +1,112 @@
...
-                        </form>
-                        @if($canDelete === true)
-                            <form action="" name="form-edit" method="DELETE">
-                                <button type="submit" class="btn btn-danger">Delete</button>
+    <table class="table table-striped">
+        <thead>
+            <tr>
+                <th scope="col">Id</th>
+                <th scope="col">Title</th>
+                <th scope="col">Action</th>
```
Comment (ToaiLC) [inline resources/views/blog/index.blade.php]: why do you have 2 col same name Action?

File: app/Http/Controllers/User/BlogController.php (hunk 9 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: I fixed

File: app/Http/Controllers/User/BlogController.php (hunk 8 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: i fixed

File: resources/views/blog/index.blade.php (hunk 10 above)
Comment (nguyentienthinh) [reply resources/views/blog/index.blade.php]: I fixed

File: app/Http/Controllers/User/BlogController.php [hunk 11]
```diff
@@ -5,14 +5,16 @@
...
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Auth;
 
 class BlogController extends Controller
 {
 
     // return information user
-    public function defineUser() {
+    public function defineUser()
+    {
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php]: why have this format changing ?

File: app/Http/Controllers/User/BlogController.php [hunk 12]
```diff
@@ -23,17 +25,14 @@ public function defineUser() {
...
-        if ($this->defineUser()->can_delete === Blog::IS_TRUE) {
-            $canDelete = true;
-        }else {
-            $canDelete = false;
-        }
+        $blogs = Blog::where('user_id', '=', $this->defineUser()->id)->get();
+        $canDelete = $this->defineUser()->can_delete === Blog::IS_TRUE;
         $canSee = $this->defineUser()->can_see;
-        if($canSee === Blog::IS_TRUE)
+        if ($canSee === Blog::IS_TRUE) {
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php]: why have this format changing ?

File: app/Http/Controllers/User/BlogController.php [hunk 13]
```diff
@@ -23,17 +25,14 @@ public function defineUser() {
...
             return view('blog/index', ['blogs'=>$blogs, 'canDelete'=>$canDelete]);
-        else
-            return redirect()->back();
+        } else {
+            return redirect()->back()->withErrors("you can't to enter blog manage");
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php:29]: if this condition only use $canSee , move it to top of the function and do the return from there

File: app/Http/Controllers/User/BlogController.php [hunk 14]
```diff
@@ -48,11 +47,85 @@ public function store(Request $request)
...
+        }
+    }
+
+    /**
+     * Delete one blog.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php:71]: @param not match

File: app/Http/Controllers/User/BlogController.php [hunk 15]
```diff
@@ -48,11 +47,85 @@ public function store(Request $request)
...
+                    $blog->delete();
+                } catch (Exception $e) {
+                    return redirect()->back()->withErrors('Delete blog error');
+                }
+            } else {
+                return redirect()->back()->withErrors('Delete blog error');
+            }
+        }
+
+        return redirect()->back()->with('message', 'Delete blog success !');
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php:90]: create a small if with direct return at begin of the function is better than a big if then small return

File: app/Http/Controllers/User/BlogController.php [hunk 16]
```diff
@@ -48,11 +47,85 @@ public function store(Request $request)
...
+    }
+
+    /**
+     * Edit view blog.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php:99]: @param not match

File: app/Http/Controllers/User/BlogController.php [hunk 17]
```diff
@@ -48,11 +47,85 @@ public function store(Request $request)
...
+    {
+        $blog = Blog::find($id);
+        if (isset($blog)) {
+            return view('blog/edit', ['blog' => $blog]);
+        } else {
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php]: with if + return , code can write directly no need the else

File: app/Http/Controllers/User/BlogController.php [hunk 18]
```diff
@@ -48,11 +47,85 @@ public function store(Request $request)
...
+        }
+    }
+
+    /**
+     * Update one blog.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php:116]: @param not match

File: app/Http/Controllers/User/BlogController.php [hunk 19]
```diff
@@ -48,11 +47,85 @@ public function store(Request $request)
...
+                $blog->content = $request->content;
+                $blog->save();
+            } catch (Exception $e) {
+                return redirect()->back()->withErrors('Update blog error');
+            }
+
+            return redirect()->route('blog.index')->with('message', 'Update blog success!');
+        }
+
+        return redirect()->back()->withErrors('Update blog error!');
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php]: this case can go with small if direct return , better than big if long wait till return

**Review by hienl — CHANGES_REQUESTED** [review]
Please Fix

File: app/Http/Controllers/User/BlogController.php (hunk 11 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: Dạ em check phpcs thi bị lỗi, em viết lại theo format PSR2.

File: app/Http/Controllers/User/BlogController.php (hunk 12 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: Dạ em check phpcs thi bị lỗi, em viết lại theo format PSR2.

File: app/Http/Controllers/User/BlogController.php (hunk 14 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php:71]: I fixed

File: app/Http/Controllers/User/BlogController.php (hunk 16 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php:99]: I fixed

File: app/Http/Controllers/User/BlogController.php (hunk 17 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: I fixed

File: app/Http/Controllers/User/BlogController.php (hunk 18 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php:116]: i fixed

File: app/Http/Controllers/User/BlogController.php (hunk 11 above)
Comment (hienl) [reply app/Http/Controllers/User/BlogController.php]: trả lại như cũ (vì ko phải code sửa trong lần này)

File: app/Http/Controllers/User/BlogController.php (hunk 12 above)
Comment (hienl) [reply app/Http/Controllers/User/BlogController.php]: trả lại như cũ (vì ko phải code sửa trong lần này)

File: app/Http/Controllers/User/BlogController.php (hunk 11 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: I fixed

File: app/Http/Controllers/User/BlogController.php (hunk 12 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: I fixed

Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php]: I fixed

File: app/Http/Controllers/User/BlogController.php (hunk 15 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php:90]: I fixed

File: app/Http/Controllers/User/BlogController.php [hunk 20]
```diff
@@ -23,17 +24,17 @@ public function defineUser() {
...
-            $canDelete = false;
-        }
         $canSee = $this->defineUser()->can_see;
-        if($canSee === Blog::IS_TRUE)
+        if ($canSee === Blog::IS_FALSE) {
+            return redirect()->back()->withErrors("you can't to enter blog manage");
+        }
+
+        //return view blog index with param $candelete and $blog
+        if ($canSee === Blog::IS_TRUE) {
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php]: remove this 'if' , it is no need now

File: app/Http/Controllers/User/BlogController.php [hunk 21]
```diff
@@ -23,17 +24,17 @@ public function defineUser() {
...
+            $canDelete = $this->defineUser()->can_delete === Blog::IS_TRUE;
+            $blogs = Blog::where('user_id', '=', $this->defineUser()->id)->get();
             return view('blog/index', ['blogs'=>$blogs, 'canDelete'=>$canDelete]);
```
Comment (hienl) [inline app/Http/Controllers/User/BlogController.php]: add 1 empty line before last return

**Review by hienl — CHANGES_REQUESTED** [review]
Please Fix

**Review by hienl — APPROVED** [review]
LGTM

**Pull request merged**